### PR TITLE
Create project.yaml, a source-of-truth for releasable units and test distros

### PR DESCRIPTION
**Pull request by Subbarker**
## Description

To reduce configs duplication, create project.yaml. This is foremost to unblock a reduction in Kokoro test configs, a blocker on Louhi migration. 

## Checklist:
- Unit tests
  - [x] Unit tests do not apply.
  - [ ] Unit tests have been added/modified and passed for this PR.
- Integration tests
  - [ ] Integration tests do not apply.
  - [x] Integration tests have been added/modified and passed for this PR.
- Documentation
  - [x] This PR introduces no user visible changes.
  - [ ] This PR introduces user visible changes and the corresponding documentation change has been made.
- Minor version bump
  - [x] This PR introduces no new features.
  - [ ] This PR introduces new features, and there is a separate PR to bump the [minor version](https://github.com/GoogleCloudPlatform/ops-agent/blob/master/VERSION) since the last [release](https://github.com/GoogleCloudPlatform/ops-agent/releases) already.
  - [ ] This PR bumps the version.

### CODE REVIEW

File: kokoro/scripts/test/go_test.sh [hunk 1]
```diff
@@ -53,6 +53,45 @@ track_flakiness
 # Avoids "fatal: detected dubious ownership in repository" errors on Kokoro containers.
 git config --global --add safe.directory "$(pwd)"
 
+# A helper for parsing YAML files.
+# Ex: VALUE=$(yaml ~/my_yaml_file.yaml "['a_key']")
+function yaml() {
+  python3 -c "import yaml;f=yaml.safe_load(open('$1'))$2;print(','.join(str(i) for i in f) if type(f)==list else f);"
```
Comment (martijnvans) [inline kokoro/scripts/test/go_test.sh:59]: I feel like we can make this more readable by introducing line breaks and renaming f and i:

```
function yaml() {
  python3 -c "import yaml
data=yaml.safe_load(open('$1'))$2
if type(data)==list:
  print(','.join(str(elem) for elem in data))
else:
 print(data)"
}
```

File: kokoro/scripts/test/go_test.sh [hunk 2]
```diff
@@ -53,6 +53,45 @@ track_flakiness
...
+}
+
+function set_platforms() {
+  # if PLATFORMS is defined, do nothing
+  if [ -n "${PLATFORMS}" ]; then
```
Comment (martijnvans) [inline kokoro/scripts/test/go_test.sh:64]: go/shell-style#tests says to use `[[` instead of `[`.

File: kokoro/scripts/test/go_test.sh [hunk 3]
```diff
@@ -53,6 +53,45 @@ track_flakiness
...
+    return 0
+  fi
+  # if _LOUHI_TAG_NAME is defined, set TARGET and ARCH env vars by parsing it.
+  # Example value: louhi/2.46.0/shortref/windows/x86_64/start
+  if [ -n "${_LOUHI_TAG_NAME}" ]; then
+    local split_tag=(${_LOUHI_TAG_NAME//\// })
```
Comment (martijnvans) [inline kokoro/scripts/test/go_test.sh:70]: This is kinda hard to read and weird (replace `/` by space, slam it into an array using `()`) and won't work if there are ever any spaces in these tags. What about:

```
TARGET="$(echo -n "${_LOUHI_TAG_NAME} | cut --delimiter="/" --fields=4)"
ARCH="$(echo -n "${_LOUHI_TAG_NAME} | cut --delimiter="/" --fields=5)"
```

adapted from https://stackoverflow.com/a/29903172.

File: kokoro/scripts/test/go_test.sh [hunk 4]
```diff
@@ -53,6 +53,45 @@ track_flakiness
...
+  # if ARCH is not set, return an error
+  if [ -z "${ARCH}" ]; then
+    echo "If TARGET is set, ARCH must be as well." 1>&2
+    return 1
+  fi
+  # At minimum, PLATFORMS will be the distros from "representative" for TARGET/ARCH in projects.yaml.
+  local platforms=$(yaml project.yaml "['targets']['${TARGET}']['architectures']['${ARCH}']['representative']")
+  # If not a presubmit job, add the exhaustive list of test distros.
+  if ["${KOKORO_ROOT_JOB_TYPE:-$KOKORO_JOB_TYPE}" != PRESUBMIT_*]; then
+    platforms=${platforms},$(yaml project.yaml "['targets']['${TARGET}']['architectures']['${ARCH}']['exhaustive']") || true
```
Comment (martijnvans) [inline kokoro/scripts/test/go_test.sh:88]: This is going to double-count things that are both in `representative` and `exhaustive`. They will be tested twice.

File: project.yaml [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+      x86_64:
+        test_distros:
+          representative:
+            rocky-linux-8
+          exhaustive:
+            rhel-8
+            rhel-8-2-sap-ha
+            rhel-8-4-sap-ha
+            rhel-8-6-sap-ha
+    rockylinux9
```
Comment (martijnvans) [inline project.yaml:98]: this looks like a parse error, weird indent and missing `:`

File: kokoro/scripts/test/go_test.sh (hunk 3 above)
Comment (Subbarker) [reply kokoro/scripts/test/go_test.sh:70]: Done

File: kokoro/scripts/test/go_test.sh (hunk 4 above)
Comment (Subbarker) [reply kokoro/scripts/test/go_test.sh:88]: The yaml is structured without overlaps.

File: kokoro/scripts/test/go_test.sh (hunk 1 above)
Comment (Subbarker) [reply kokoro/scripts/test/go_test.sh:59]: Done

File: project.yaml (hunk 5 above)
Comment (Subbarker) [reply project.yaml:98]: Done

File: kokoro/scripts/test/go_test.sh (hunk 2 above)
Comment (Subbarker) [reply kokoro/scripts/test/go_test.sh]: Done

**Review by martijnvans — APPROVED** [review]

File: kokoro/scripts/test/go_test.sh (hunk 4 above)
[the remark this answers is not in the document]
Comment (Subbarker) [reply kokoro/scripts/test/go_test.sh:92]: Done

File: kokoro/scripts/test/go_test.sh [hunk 6]
```diff
@@ -53,6 +53,45 @@ track_flakiness
...
+  fi
+  # if ARCH is not set, return an error
+  if [ -z "${ARCH}" ]; then
+    echo "If TARGET is set, ARCH must be as well." 1>&2
+    return 1
+  fi
+  # At minimum, PLATFORMS will be the distros from "representative" for TARGET/ARCH in projects.yaml.
+  local platforms=$(yaml project.yaml "['targets']['${TARGET}']['architectures']['${ARCH}']['representative']")
+  # If not a presubmit job, add the exhaustive list of test distros.
+  if ["${KOKORO_ROOT_JOB_TYPE:-$KOKORO_JOB_TYPE}" != PRESUBMIT_*]; then
```
[the remark this answers is not in the document]
Comment (Subbarker) [reply kokoro/scripts/test/go_test.sh]: Done

**Pull request merged**